index.js

```diff
@@ -1,53 +1,42 @@
 const dogs = []
-document.addEventListener('DOMContentLoaded', () => {
-    fetch('http://localhost:3000/dogs') 
+document.addEventListener("DOMContentLoaded", () => {
+    fetch("http://localhost:3000/dogs")
     .then(response => response.json())
     .then(data => getDogs(data))
    
-    const searchForm = document.getElementById("search-form");
+    const searchForm = document.getElementById("search-form")
     searchForm.addEventListener("submit", (e) => {searchDogs(e)})    
    
     const allDogsButton = document.getElementById("all")
-    allDogsButton.addEventListener("click", () => {showAllDogs()})
-    
+    allDogsButton.addEventListener("click", () => {
+        const dc = document.getElementById("dogs-container")
+        if (dc.children.length == 0) {
+            showAllDogs()
+        }
+    })
 })
 
-function getDogs(data) {
-    data.forEach((dog) => dogs.push(dog))
-    
+function getDogs(dogData) {
```

Comment:
This is also minor! You passed the variable "data" to this function, and I wanted to demonstrate two things:

1. You can change the variable name between functions (if it makes sense to do so).
2. "Data" is vague, especially when you have an app that has multiple fetches, so it's good to be specific.